obal/data/roles/build_package/tasks/copr_build.yml

```diff
@@ -0,0 +1,120 @@
+        project: "{{ copr_project['copr_project_name'] }}"
+        nvr: "{{ package_nvr.nvr }}"
+        package: "{{ package_nvr.name }}"
+        config_file: "{{ build_package_copr_config | default(omit) }}"
+      register: build_exists
+
+- debug:
+    msg: "{{ build_exists.skipped is defined }}"
+
+- name: Create temporary build directory
+  tempfile:
+    state: directory
+    suffix: srpms
+  register: srpm_directory
+  when: (srpm_directory is not defined and (build_exists.exists is defined and not build_exists.exists)) or build_package_copr_rebuild
+
+- name: 'Build SRPM'
+  srpm:
+    package: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
+    output: "{{ srpm_directory.path if 'path' in srpm_directory else srpm_directory }}"
+    source_location: "{{ source_location | default(omit) }}"
+    source_system: "{{ source_system | default(omit) }}"
+  register: srpm_build
+  when: (srpm_directory is defined and (build_exists.exists is defined and not build_exists.exists)) or build_package_copr_rebuild or build_package_scratch
+
+- when: (not build_package_scratch and (build_exists.exists is defined and not build_exists.exists)) or build_package_copr_rebuild
+  block:
+    - name: 'Run build'
+      copr_build:
+        user: "{{ copr_project_user }}"
+        project: "{{ copr_project['copr_project_name'] }}"
+        srpm: "{{ srpm_build.path }}"
+        chroots: "{{ build_package_copr_chroots | default(omit) }}"
+        config_file: "{{ build_package_copr_config | default(omit) }}"
+      register: copr_builds
+      ignore_errors: "{{ build_package_skip_failed_build | default(false) }}"
+
+    - name: 'Created tasks'
+      debug:
+        var: copr_builds.build_urls
+      when: copr_builds.changed
+
+    - name: 'Wait for tasks to finish'
+      include_tasks: wait.yml
+      when: build_package_wait|bool and copr_builds.changed
+      vars:
+        copr_build_ids: "{{ copr_builds.builds }}"
+
+- when: build_package_scratch
+  block:
+    - name: Define Copr scratch project name
+      set_fact:
+        copr_scratch_project: "{{ copr_project['copr_project_name'] }}-scratch-{{ 999999999 | random | to_uuid }}"
+      when: copr_scratch_project is not defined
+
+    - name: Define Copr scatch user name
+      set_fact:
+        copr_scratch_user: "{{ copr_project_user }}"
+      when: copr_scratch_user is not defined
+
+    - include_role:
+        name: copr_project
+      vars:
+        copr_project_user: "{{ copr_scratch_user }}"
+        copr_project_name: "{{ copr_scratch_project }}"
+        copr_project_chroots: "{{ copr_project['copr_project_chroots'] }}"
+        copr_project_description: "{{ copr_project['copr_project_description'] | default(omit) }}"
+        copr_project_delete_after_days: "4"
+        copr_project_unlisted_on_homepage: "{{ copr_project['copr_project_unlisted_on_homepage'] | default(omit) }}"
+        copr_project_copr_config: "{{ build_package_copr_config | default(omit) }}"
```

Comment:
you can't pass `default(omit)` to roles, only to modules.

Reply:
Will this properly handle undefined variables? Or is default(None) needed?

Reply:
I am afraid no. No idea if `default(None)` or `default(~)` will work, try it :)